### PR TITLE
[TF] Remove security Checker

### PR DESCRIPTION
**Pull request by nmittler**
**Please provide a description of this PR:**

**To help us figure out who should review this PR, please put an X in all the areas that this PR affects.**

- [ ] Configuration Infrastructure
- [ ] Docs
- [ ] Installation
- [ ] Networking
- [ ] Performance and Scalability
- [ ] Policies and Telemetry
- [ ] Security
- [x] Test and Release
- [ ] User Experience
- [ ] Developer Infrastructure

**Please check any characteristics that apply to this pull request.**

- [ ] Does not have any [user-facing](https://github.com/istio/istio/tree/master/releasenotes#when-to-add-release-notes) changes. This may include CLI changes, API changes, behavior changes, performance improvements, etc.

### CODE REVIEW

File: tests/integration/security/ca_custom_root/secure_naming_test.go [hunk 1]
```diff
@@ -187,17 +184,12 @@ func TestSecureNaming(t *testing.T) {
...
-									From:          a,
-									Options:       callOptions,
-									ExpectSuccess: tc.expectSuccess,
-									DestClusters:  bSet.Clusters(),
-								}
-								if err := retry.UntilSuccess(
-									checker.Check, retry.Delay(time.Second), retry.Timeout(15*time.Second), retry.Converge(5)); err != nil {
-									t.Fatal(err)
-								}
+								a.CallWithRetryOrFail(t, callOptions, retry.Converge(5))
```
Comment (howardjohn) [inline tests/integration/security/ca_custom_root/secure_naming_test.go:192]: nit: the default is now converge(3) so we probably do not need converge(5)

**Review by howardjohn — COMMENTED** [review]
nice

File: tests/integration/security/ca_custom_root/secure_naming_test.go (hunk 1 above)
Comment (nmittler) [reply tests/integration/security/ca_custom_root/secure_naming_test.go:191]: It's actually not an echo default currently. I think it probably should be though. Let me create a PR to change it.

File: tests/integration/security/authorization_test.go [hunk 2]
```diff
@@ -1450,113 +1378,221 @@ extensionProviders:
...
+		echo.ValidatorFunc(func(resp client.ParsedResponses, err error) error {
+			if err == nil {
+				err = resp.CheckOK()
+			}
+			if err != nil {
+				return getError(from, opts, "allow with code 200", fmt.Sprintf("error: %v", err))
+			}
+			return nil
+		}),
+		validators.ExpectReachedClusters(to, opts))
```
Comment (stevenctl) [inline tests/integration/security/authorization_test.go:1554]: Not sure why yet, but I saw this test breaking for multi-network. If this PR doesn't somehow fix it: 

```suggestion
		validators.ExpectReachedClusters(to.Clusters().ByNetwork()[from.Config().Cluster.NetworkName()], opts))
```

Comment (stevenctl) [reply tests/integration/security/authorization_test.go]: In a ~recent attempt to enable this for multi-cluster it was broken for multi-network.

File: tests/integration/security/ca_custom_root/multi_root_test.go [hunk 3]
```diff
@@ -37,46 +38,41 @@ func TestMultiRootSetup(t *testing.T) {
...
 								Scheme:   s,
 							}
-							checker := connection.Checker{
-								From:          src,
-								Options:       opt,
-								ExpectSuccess: success,
-								DestClusters:  t.Clusters(),
-							}
-							checker.CheckOrFail(t)
+							opts.Check = check.And(check.OK(), scheck.ReachedClusters(to, &opts))
```
Comment (howardjohn) [inline tests/integration/security/ca_custom_root/multi_root_test.go:55]: super nit: define inline in the struct creations

File: tests/integration/security/ca_custom_root/multi_root_test.go [hunk 4]
```diff
@@ -37,46 +38,41 @@ func TestMultiRootSetup(t *testing.T) {
...
+						name := fmt.Sprintf("server:%s[%s]", to[0].Config().Service, want)
 						ctx.NewSubTest(name).Run(func(t framework.TestContext) {
 							t.Helper()
-							opt := echo.CallOptions{
-								Target:   dest,
+							opts := echo.CallOptions{
+								Target:   to[0],
 								PortName: HTTPS,
-								Address:  dest.Config().Service,
+								Address:  to[0].Config().Service,
```
Comment (howardjohn) [inline tests/integration/security/ca_custom_root/multi_root_test.go:52]: very nit: not needed, Target implies this

**Review by howardjohn — APPROVED** [review]

File: tests/integration/security/ca_custom_root/multi_root_test.go (hunk 3 above)
Comment (nmittler) [reply tests/integration/security/ca_custom_root/multi_root_test.go:55]: Can't ... the `ReachedClusters` requires the struct.

**Pull request merged**